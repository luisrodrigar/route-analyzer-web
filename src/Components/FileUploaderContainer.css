.FileUploaderContainer{
	padding:15px;
}

.FileUploaderForm select, .FileUploaderForm input{
	padding:10px;
}

.FileUploaderForm input[type="submit"]{
	padding:10px;
}