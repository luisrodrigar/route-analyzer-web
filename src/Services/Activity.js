
import {Exception} from './Exception'
import axios from 'axios';

const DIR_BASE = "https://routeanalyzer-api.herokuapp.com/";

export function get(id){
	const path = DIR_BASE + "/activity/" + id ;
	return axios
		.get(path)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception("Not match any activity with the id passed as a parameter.");
    	});
}

export function exportAs(id, type){
	const path = DIR_BASE + "/activity/" + id + "/export/" + type ;
	return axios
		.get(path)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception("Problem trying to export the file as " + type);
    	});
}

export function joinLaps(id, dataSelected){
	let indexLap1 = dataSelected[0].indexLap - 1;
	let indexLap2 = dataSelected[1].indexLap - 1;

	const url = DIR_BASE + "/activity/" + id +"/join/laps?index1=" 
		+ indexLap1 + "&index2=" + indexLap2;

	return axios
		.put(url)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception(err.response.data.description);
    	});
}

export function removePoint(id, position, timeInMillis, index){
	const positionParam = getPositionParams(position);
	const time = !timeInMillis || isNaN(timeInMillis) ? null : "timeInMillis=" + timeInMillis ;
	const indexParam = "index="+index;
	
	const url = DIR_BASE + "/activity/" + id +"/remove/point?" 
		+ positionParam + (time ? "&" + time : "&timeInMillis=") + "&" + indexParam;

	return axios
		.put(url)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception(err.response.data.description);
    	});
}

export function splitLap(id, position, timeInMillis, index){
	const positionParam = getPositionParams(position);
	const time = !timeInMillis || isNaN(timeInMillis) ? null : "timeInMillis=" + timeInMillis ;
	const indexParam = "index="+index;
	
	const url = DIR_BASE + "/activity/" + id +"/split/lap?" + positionParam + "&" + indexParam + (time ? "&" + time : "&timeInMillis=");

	return axios
		.put(url)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception(err.response.data.description);
    	});
}

export function removeLaps(id,dataSelected){

	const indexLaps = dataSelected.map(lap => {
		return lap.indexLap;
	}).join(",");

	const startTimeLaps = dataSelected.map(lap => {
		return lap.startTime?lap.startTime:null;
	}).join(",");

	const url = DIR_BASE + "/activity/" + id +"/remove/laps?date=" + startTimeLaps + "&index=" + indexLaps;

	return axios
		.put(url)
    	.then(res => res.data)
    	.catch(err => {
    		throw new Exception(err.response.data.description);
    	});
}

export function setColors(id, data){
	const url = DIR_BASE + "/activity/" + id +"/color/laps?data=" + data;
	return axios
		.put(url)
		.then()
    	.catch(err => {
    		throw new Exception("It cannot be posible to set the colors");
    	});
}

function getPositionParams(position){
	return "lat="+position.lat+"&lng="+position.lng;
}