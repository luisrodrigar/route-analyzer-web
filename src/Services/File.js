import axios from 'axios';
import Exception from '../Exception'

export function uploadFile(formData) {
   this.path = "http://localhost:8080/RouteAnalyzer/file/upload";
    axios
      .post(this.path, formData)
      .then(res => { 
        return res.data[0]; 
      } )
      .catch(err =>{
        throw new Exception(err.response.data.description);
      });
}



export default File;