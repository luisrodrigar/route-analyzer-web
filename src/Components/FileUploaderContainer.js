import React, { Component } from 'react';
import {upload} from '../Services/file';
import './FileUploaderContainer.css';


class FileUploaderContainer extends Component {

  constructor(props){
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.showProgressIcon();
    const formData = new FormData();
    formData.append("name", "name de ejemplo");
    formData.append('file', this.fileInput.files[0]);
    formData.append('type', this.type.value);
    
    upload(formData)
      .then(idActivity => {
        this.props.showMapRoute(idActivity);
      })
      .catch(err => {
        alert(err.message);
      });

  };

  render() {
    return (
      <div className="FileUploaderContainer">
      <form className="FileUploaderForm" onSubmit={this.handleSubmit}>
        <input 
          type="file" 
          ref={input => { this.fileInput = input; }} />
        <select defaultValue="gpx" className="optionFileUploaderForm" ref={option => { this.type = option; }} >
          <option value='gpx'>gpx</option>
          <option value='tcx'>tcx</option>
        </select>
        <input type="submit" value="Submit" />
      </form>
      </div>
    );
  }
}

export default FileUploaderContainer;