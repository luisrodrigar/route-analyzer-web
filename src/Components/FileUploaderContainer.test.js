import React from 'react';
import { shallow, mount } from 'enzyme';
import { FileUploaderContainer } from './FileUploaderContainer';
jest.mock("../Services/file");
import * as fileServices from '../Services/file';

// Create the mock store
import configureMockStore from 'redux-mock-store';
import toJson from "enzyme-to-json";
const mockStore = configureMockStore();


describe('File Updater Container Test Cases', () => {
    let preventDefaultMock;
    let appendMockFn;
    let toggleProgressMock;
    let actionShowRouteMock;
    let activityId;
    beforeAll(() => {
        activityId = "5ace8cd14c147400048aa6b0";
    });
    beforeEach(() => {
        toggleProgressMock = jest.fn();
        actionShowRouteMock = jest.fn();
        appendMockFn = jest.fn();
        window.FormData = jest.fn(() => ({ append: appendMockFn }));
    });
    afterEach(() => {
        jest.resetAllMocks();
    });
    it('Render initialization successfully', () => {
        const component = shallow(<FileUploaderContainer toggleProgress={toggleProgressMock}
                                                       actionShowRoute={actionShowRouteMock} />);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('FormData is called with right tcx param and file.', async () => {
        const component = mount(<FileUploaderContainer toggleProgress={toggleProgressMock}
                                                       actionShowRoute={actionShowRouteMock} />);
        fileServices.upload.mockResolvedValueOnce(jest.fn());
        const mockFileName = "Mock file name";
        const fileInputMock = {
            files: [mockFileName]
        };
        component.instance().fileInput = fileInputMock;
        const tcxTypeMock = {
            value: "tcx"
        }
        component.instance().type = tcxTypeMock;

        preventDefaultMock = {
            preventDefault: jest.fn(() => ({}))
        };
        const form = component.find('form');
        await form.simulate('submit', preventDefaultMock);
        expect(appendMockFn).toBeCalledTimes(2);
        expect(appendMockFn).toBeCalledWith("file", mockFileName);
        expect(appendMockFn).toBeCalledWith("type", tcxTypeMock.value);

        const formData = new FormData();
        formData.append('file', mockFileName);
        formData.append('type', tcxTypeMock.value);
        expect(fileServices.upload).toBeCalledWith(formData);
    });
    it('Handling submit file.', async () => {
        const component = mount(<FileUploaderContainer toggleProgress={toggleProgressMock}
                                                         actionShowRoute={actionShowRouteMock} />);
        fileServices.upload.mockResolvedValueOnce(activityId);
        preventDefaultMock = {
            preventDefault: jest.fn(() => ({}))
        };
        const form = component.find('form');
        await form.simulate('submit', preventDefaultMock);

        expect(preventDefaultMock.preventDefault).toBeCalled();
        expect(toggleProgressMock).toBeCalledWith(true);
        expect(fileServices.upload).toBeCalled();
        expect(actionShowRouteMock).toBeCalledWith("5ace8cd14c147400048aa6b0");
        expect(actionShowRouteMock).not.toBeCalledWith(null);
    });
    it('Error trying to upload some file', async () => {
        const component = mount(<FileUploaderContainer toggleProgress={toggleProgressMock}
                                                       actionShowRoute={actionShowRouteMock} />);
        window.alert = jest.fn(() => ({}));

        const error = {
            message: "Error mocked"
        };
        fileServices.upload.mockRejectedValueOnce(jest.fn(() => error));
        preventDefaultMock = {
            preventDefault: jest.fn(() => ({}))
        };
        const form = component.find('form');
        form.simulate('submit', preventDefaultMock);

        expect(preventDefaultMock.preventDefault).toBeCalled();
        expect(toggleProgressMock).toBeCalledWith(true);
        expect(fileServices.upload).toBeCalled();
        expect(actionShowRouteMock).not.toBeCalledWith("5ace8cd14c147400048aa6b0");
        // expect(await window.alert).toBeCalledWith(error.message);
        // expect(await actionShowRouteMock).toBeCalledWith(null);
    });
});
