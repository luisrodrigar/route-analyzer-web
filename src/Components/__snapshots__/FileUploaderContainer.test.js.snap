// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Updater Container Test Cases Render initialization successfully 1`] = `
<div
  className="FileUploaderContainer"
>
  <form
    className="FileUploaderForm"
    onSubmit={[Function]}
  >
    <input
      type="file"
    />
    <select
      className="optionFileUploaderForm"
      defaultValue="gpx"
    >
      <option
        value="gpx"
      >
        gpx
      </option>
      <option
        value="tcx"
      >
        tcx
      </option>
    </select>
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
